File: projects/solms/training/courses/c++/book/OOIntro/Programs/Account1.cpp
\begin{verbatim}
class Account
{
  public:
    void credit(double amount) {_balance += amount;}
    void debit(double amount) {_balance -= amount;}
    double balance() {return _balance;}

  private:
    double _balance;
};

#include <iostream>

using namespace std;

int main()
{
  Account acc1;
  cout << "balance = " << acc1.balance() << endl;

  acc1.credit(10000);
  cout << "balance = " << acc1.balance() << endl;

  char c; cin >> c;
  
  return 0;
}
\end{verbatim}

File: resources/information/technologies/java/lang/classes/basic/this/src/Account1.java
public class Account1
{
   public Account1(String accountNumber)
   {
     this( accountNumber, 0.0 );
   }
 
   public Account1(String accountNumber, double initialBalance)
   {
     accountNo = accountNumber;
     balance   = initialBalance;
   }
 
   public void credit(double amount)
   {
     balance += amount;
   }
 
   public void debit(double amount)
   {
     credit(-amount);   
     // equivalent to this.credit(-amount);
   }
 
   public double getBalance()
   {
     return balance;
   }
 
   public String getAccountNumber()
   {
    return accountNo;
   }
 
   public String toString()
   {
     return accountNo + ": " + balance;
   }
 
   private String accountNo;
   private double balance;
}
